lib/instrumentation/http-shared.js

```diff
@@ -152,23 +153,25 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
       // If so, then fallback to the current run context's span or transaction,
       // if any.
       var parent = span || parentRunContext.currSpan() || parentRunContext.currTransaction()
-      if (parent && parent._context) {
-        const headerValue = parent._context.toTraceParentString()
-        const traceStateValue = parent._context.toTraceStateString()
-
-        const newHeaders = Object.assign({}, options.headers)
-        newHeaders.traceparent = headerValue
-        if (traceStateValue) {
-          // Do not set the header to the empty string. Though RFC 7230 allows
-          // empty header values, some servers apparently do not.
-          newHeaders.tracestate = traceStateValue
+
+      const setter = function (carrier, name, value) {
+        if (!value) {
+          return
         }
```

Comment:
Instead of checking for empty values in the setter, can TraceContext guarantee to only call the setter for non-empty values? This makes the life of all setters easier.

Reply:
Seems reasonable -- done in 587e1d4395071a34f0604d22300e9be790a52e3c.